clifford/_mvarray.py

```diff
@@ -1,57 +1,96 @@
-import numpy as np
+import functools
+import operator
+from typing import Tuple
 
+import numpy as np
 
 from clifford.io import write_ga_file, read_ga_file  # noqa: F401
+from ._layout import Layout
 from ._multivector import MultiVector
 
 dual_array = np.vectorize(MultiVector.dual)
 normal_array = np.vectorize(MultiVector.normal)
 call_array = np.vectorize(MultiVector.__call__)
 
 
+def _interrogate_nested_mvs(input_array) -> Tuple[Tuple[int], Layout, np.dtype]:
+    """
+    Calculates the shape of the nested input_array, and gets the associated layout.
+    Stops descending when it encounters a MultiVector.
+    """
+    if not isinstance(input_array, MultiVector):
+        nested_shape, layout, dtype = _interrogate_nested_mvs(input_array[0])
+        return (len(input_array), *nested_shape), layout, dtype
+    else:
+        return (), input_array.layout, input_array.value.dtype
+
+
+def _index_nested_iterable(input_iterable, index):
+    """
+    Given a nested iterable, input_iterable, return the element given by the
+    1d index iterable
+    """
+    return functools.reduce(operator.getitem, index, input_iterable)
+
+
+@functools.lru_cache(None)
+def _get_vectorized_value_func(dtype):
+    return np.vectorize(lambda x: x.value, otypes=[dtype], signature='()->(n)')
+
+
 class MVArray(np.ndarray):
-    '''
+    """
     MultiVector Array
-    '''
-
+    """
     def __new__(cls, input_array):
-        obj = np.empty(len(input_array), dtype=object)
-        obj[:] = input_array
-        obj = obj.view(cls)
-        return obj
+
+        input_shape, layout, dtype = _interrogate_nested_mvs(input_array)
+        # copying this across elementwise is necessary to prevent numpy recursing into the multivector coefficients
+        obj = np.empty(input_shape, dtype=object)
+        for index in np.ndindex(input_shape):
+            obj[index] = _index_nested_iterable(input_array, index)
+
+        self = obj.view(cls)
+        return self
 
     def __array_finalize__(self, obj):
         if obj is None:
             return
 
+    def _get_first_element(self):
+        return _index_nested_iterable(self, [0] * len(self.shape))
+
     @property
     def value(self):
         """
         Return an np array of the values of multivectors
         """
-        return np.array([mv.value for mv in self])
+        value_dtype = self._get_first_element().value.dtype
+        return _get_vectorized_value_func(value_dtype)(self)
```

Comment:
If the test I suggest below doesn't pass, this will need to become
```suggestion
        return _get_vectorized_value_func(value_dtype)(self).view(np.ndarray)
```